// ng2-bs3-modal/src/ng2-bs3-modal/components/modal.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var import1 = require("@angular/core/src/linker/view");
var import2 = require("@angular/core/src/linker/element");
var import3 = require("./modal");
var import4 = require("@angular/core/src/linker/view_utils");
var import6 = require("@angular/core/src/linker/view_type");
var import7 = require("@angular/core/src/change_detection/change_detection");
var import8 = require("@angular/core/src/linker/element_ref");
var import9 = require("@angular/core/src/metadata/view");
var import10 = require("@angular/core/src/linker/component_factory");
var import11 = require("@angular/common/src/directives/ng_class");
var import12 = require("@angular/core/src/change_detection/differs/iterable_differs");
var import13 = require("@angular/core/src/change_detection/differs/keyvalue_differs");
var renderType_ModalComponent_Host = null;
var _View_ModalComponent_Host0 = (function (_super) {
    __extends(_View_ModalComponent_Host0, _super);
    function _View_ModalComponent_Host0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalComponent_Host0, renderType_ModalComponent_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('modal', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'modal');
        this.renderer.setElementAttribute(this._el_0, 'role', 'dialog');
        this.renderer.setElementAttribute(this._el_0, 'tabindex', '-1');
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_ModalComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._ModalComponent_0_4 = new import3.ModalComponent(new import8.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._ModalComponent_0_4, [], compView_0);
        compView_0.create(this._ModalComponent_0_4, this.projectableNodes, null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this._expr_2 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_ModalComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.ModalComponent) && (0 === requestNodeIndex))) {
            return this._ModalComponent_0_4;
        }
        return notFoundResult;
    };
    _View_ModalComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this._ModalComponent_0_4.fadeClass;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_0, 'fade', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this._ModalComponent_0_4.dataKeyboardAttr;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementAttribute(this._el_0, 'data-keyboard', ((currVal_1 == null) ? null : currVal_1.toString()));
            this._expr_1 = currVal_1;
        }
        var currVal_2 = this._ModalComponent_0_4.dataBackdropAttr;
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementAttribute(this._el_0, 'data-backdrop', ((currVal_2 == null) ? null : currVal_2.toString()));
            this._expr_2 = currVal_2;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_ModalComponent_Host0.prototype.destroyInternal = function () {
        this._ModalComponent_0_4.ngOnDestroy();
    };
    return _View_ModalComponent_Host0;
}(import1.AppView));
function viewFactory_ModalComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalComponent_Host === null)) {
        (renderType_ModalComponent_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_ModalComponent_Host0(viewUtils, parentInjector, declarationEl);
}
exports.ModalComponentNgFactory = new import10.ComponentFactory('modal', viewFactory_ModalComponent_Host0, import3.ModalComponent);
var styles_ModalComponent = [];
var renderType_ModalComponent = null;
var _View_ModalComponent0 = (function (_super) {
    __extends(_View_ModalComponent0, _super);
    function _View_ModalComponent0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalComponent0, renderType_ModalComponent, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'modal-dialog');
        this._NgClass_1_3 = new import11.NgClass(this.parentInjector.get(import12.IterableDiffers), this.parentInjector.get(import13.KeyValueDiffers), new import8.ElementRef(this._el_1), this.renderer);
        this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
        this._el_3 = this.renderer.createElement(this._el_1, 'div', null);
        this.renderer.setElementAttribute(this._el_3, 'class', 'modal-content');
        this._text_4 = this.renderer.createText(this._el_3, '\n                ', null);
        this.renderer.projectNodes(this._el_3, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._text_5 = this.renderer.createText(this._el_3, '\n            ', null);
        this._text_6 = this.renderer.createText(this._el_1, '\n        ', null);
        this._text_7 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([], [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4,
            this._text_5,
            this._text_6,
            this._text_7
        ], [], []);
        return null;
    };
    _View_ModalComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import11.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 6)))) {
            return this._NgClass_1_3;
        }
        return notFoundResult;
    };
    _View_ModalComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = 'modal-dialog';
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._NgClass_1_3.klass = currVal_0;
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.context.getCssClasses();
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this._NgClass_1_3.ngClass = currVal_1;
            this._expr_1 = currVal_1;
        }
        if (!throwOnChange) {
            this._NgClass_1_3.ngDoCheck();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_ModalComponent0;
}(import1.AppView));
function viewFactory_ModalComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalComponent === null)) {
        (renderType_ModalComponent = viewUtils.createRenderComponentType('/Users/doug/Source/dougludlow/ng2-bs3-modal/src/ng2-bs3-modal/components/modal.ts class ModalComponent - inline template', 1, import9.ViewEncapsulation.None, styles_ModalComponent, {}));
    }
    return new _View_ModalComponent0(viewUtils, parentInjector, declarationEl);
}
exports.viewFactory_ModalComponent0 = viewFactory_ModalComponent0;
