// ng2-bs3-modal/src/ng2-bs3-modal/components/modal-header.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var import1 = require("@angular/core/src/linker/view");
var import2 = require("@angular/core/src/linker/element");
var import3 = require("./modal-header");
var import4 = require("@angular/core/src/linker/view_utils");
var import6 = require("@angular/core/src/linker/view_type");
var import7 = require("@angular/core/src/change_detection/change_detection");
var import8 = require("./modal");
var import9 = require("@angular/core/src/metadata/view");
var import10 = require("@angular/core/src/linker/component_factory");
var import11 = require("@angular/common/src/directives/ng_if");
var import12 = require("@angular/core/src/linker/template_ref");
var renderType_ModalHeaderComponent_Host = null;
var _View_ModalHeaderComponent_Host0 = (function (_super) {
    __extends(_View_ModalHeaderComponent_Host0, _super);
    function _View_ModalHeaderComponent_Host0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalHeaderComponent_Host0, renderType_ModalHeaderComponent_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalHeaderComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('modal-header', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_ModalHeaderComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._ModalHeaderComponent_0_4 = new import3.ModalHeaderComponent(this.parentInjector.get(import8.ModalComponent));
        this._appEl_0.initComponent(this._ModalHeaderComponent_0_4, [], compView_0);
        compView_0.create(this._ModalHeaderComponent_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_ModalHeaderComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.ModalHeaderComponent) && (0 === requestNodeIndex))) {
            return this._ModalHeaderComponent_0_4;
        }
        return notFoundResult;
    };
    return _View_ModalHeaderComponent_Host0;
}(import1.AppView));
function viewFactory_ModalHeaderComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalHeaderComponent_Host === null)) {
        (renderType_ModalHeaderComponent_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_ModalHeaderComponent_Host0(viewUtils, parentInjector, declarationEl);
}
exports.ModalHeaderComponentNgFactory = new import10.ComponentFactory('modal-header', viewFactory_ModalHeaderComponent_Host0, import3.ModalHeaderComponent);
var styles_ModalHeaderComponent = [];
var renderType_ModalHeaderComponent = null;
var _View_ModalHeaderComponent0 = (function (_super) {
    __extends(_View_ModalHeaderComponent0, _super);
    function _View_ModalHeaderComponent0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalHeaderComponent0, renderType_ModalHeaderComponent, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalHeaderComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'modal-header');
        this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._appEl_3 = new import2.AppElement(3, 1, this, this._anchor_3);
        this._TemplateRef_3_5 = new import12.TemplateRef_(this._appEl_3, viewFactory_ModalHeaderComponent1);
        this._NgIf_3_6 = new import11.NgIf(this._appEl_3.vcRef, this._TemplateRef_3_5);
        this._text_4 = this.renderer.createText(this._el_1, '\n            ', null);
        this.renderer.projectNodes(this._el_1, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._text_5 = this.renderer.createText(this._el_1, '\n        ', null);
        this._text_6 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._expr_0 = import7.UNINITIALIZED;
        this.init([], [
            this._text_0,
            this._el_1,
            this._text_2,
            this._anchor_3,
            this._text_4,
            this._text_5,
            this._text_6
        ], [], []);
        return null;
    };
    _View_ModalHeaderComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import11.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6;
        }
        return notFoundResult;
    };
    _View_ModalHeaderComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.context.showClose;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._NgIf_3_6.ngIf = currVal_0;
            this._expr_0 = currVal_0;
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_ModalHeaderComponent0;
}(import1.AppView));
function viewFactory_ModalHeaderComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalHeaderComponent === null)) {
        (renderType_ModalHeaderComponent = viewUtils.createRenderComponentType('/Users/doug/Source/dougludlow/ng2-bs3-modal/src/ng2-bs3-modal/components/modal-header.ts class ModalHeaderComponent - inline template', 1, import9.ViewEncapsulation.None, styles_ModalHeaderComponent, {}));
    }
    return new _View_ModalHeaderComponent0(viewUtils, parentInjector, declarationEl);
}
exports.viewFactory_ModalHeaderComponent0 = viewFactory_ModalHeaderComponent0;
var _View_ModalHeaderComponent1 = (function (_super) {
    __extends(_View_ModalHeaderComponent1, _super);
    function _View_ModalHeaderComponent1(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalHeaderComponent1, renderType_ModalHeaderComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalHeaderComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'button', null);
        this.renderer.setElementAttribute(this._el_0, 'aria-label', 'Close');
        this.renderer.setElementAttribute(this._el_0, 'class', 'close');
        this.renderer.setElementAttribute(this._el_0, 'data-dismiss', 'modal');
        this.renderer.setElementAttribute(this._el_0, 'type', 'button');
        this._text_1 = this.renderer.createText(this._el_0, '\n                ', null);
        this._el_2 = this.renderer.createElement(this._el_0, 'span', null);
        this.renderer.setElementAttribute(this._el_2, 'aria-hidden', 'true');
        this._text_3 = this.renderer.createText(this._el_2, '×', null);
        this._text_4 = this.renderer.createText(this._el_0, '\n            ', null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._text_4
        ], [disposable_0], []);
        return null;
    };
    _View_ModalHeaderComponent1.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.context.modal.dismiss() !== false);
        return (true && pd_0);
    };
    return _View_ModalHeaderComponent1;
}(import1.AppView));
function viewFactory_ModalHeaderComponent1(viewUtils, parentInjector, declarationEl) {
    return new _View_ModalHeaderComponent1(viewUtils, parentInjector, declarationEl);
}
