// ng2-bs3-modal/src/ng2-bs3-modal/components/modal-footer.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var import1 = require("@angular/core/src/linker/view");
var import2 = require("@angular/core/src/linker/element");
var import3 = require("./modal-footer");
var import4 = require("@angular/core/src/linker/view_utils");
var import6 = require("@angular/core/src/linker/view_type");
var import7 = require("@angular/core/src/change_detection/change_detection");
var import8 = require("./modal");
var import9 = require("@angular/core/src/metadata/view");
var import10 = require("@angular/core/src/linker/component_factory");
var import11 = require("@angular/common/src/directives/ng_if");
var import12 = require("@angular/core/src/linker/template_ref");
var renderType_ModalFooterComponent_Host = null;
var _View_ModalFooterComponent_Host0 = (function (_super) {
    __extends(_View_ModalFooterComponent_Host0, _super);
    function _View_ModalFooterComponent_Host0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalFooterComponent_Host0, renderType_ModalFooterComponent_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalFooterComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('modal-footer', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_ModalFooterComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._ModalFooterComponent_0_4 = new import3.ModalFooterComponent(this.parentInjector.get(import8.ModalComponent));
        this._appEl_0.initComponent(this._ModalFooterComponent_0_4, [], compView_0);
        compView_0.create(this._ModalFooterComponent_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_ModalFooterComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.ModalFooterComponent) && (0 === requestNodeIndex))) {
            return this._ModalFooterComponent_0_4;
        }
        return notFoundResult;
    };
    return _View_ModalFooterComponent_Host0;
}(import1.AppView));
function viewFactory_ModalFooterComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalFooterComponent_Host === null)) {
        (renderType_ModalFooterComponent_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_ModalFooterComponent_Host0(viewUtils, parentInjector, declarationEl);
}
exports.ModalFooterComponentNgFactory = new import10.ComponentFactory('modal-footer', viewFactory_ModalFooterComponent_Host0, import3.ModalFooterComponent);
var styles_ModalFooterComponent = [];
var renderType_ModalFooterComponent = null;
var _View_ModalFooterComponent0 = (function (_super) {
    __extends(_View_ModalFooterComponent0, _super);
    function _View_ModalFooterComponent0(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalFooterComponent0, renderType_ModalFooterComponent, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalFooterComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'modal-footer');
        this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
        this.renderer.projectNodes(this._el_1, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._text_3 = this.renderer.createText(this._el_1, '\n            ', null);
        this._anchor_4 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._appEl_4 = new import2.AppElement(4, 1, this, this._anchor_4);
        this._TemplateRef_4_5 = new import12.TemplateRef_(this._appEl_4, viewFactory_ModalFooterComponent1);
        this._NgIf_4_6 = new import11.NgIf(this._appEl_4.vcRef, this._TemplateRef_4_5);
        this._text_5 = this.renderer.createText(this._el_1, '\n            ', null);
        this._anchor_6 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._appEl_6 = new import2.AppElement(6, 1, this, this._anchor_6);
        this._TemplateRef_6_5 = new import12.TemplateRef_(this._appEl_6, viewFactory_ModalFooterComponent2);
        this._NgIf_6_6 = new import11.NgIf(this._appEl_6.vcRef, this._TemplateRef_6_5);
        this._text_7 = this.renderer.createText(this._el_1, '\n        ', null);
        this._text_8 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([], [
            this._text_0,
            this._el_1,
            this._text_2,
            this._text_3,
            this._anchor_4,
            this._text_5,
            this._anchor_6,
            this._text_7,
            this._text_8
        ], [], []);
        return null;
    };
    _View_ModalFooterComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.TemplateRef) && (4 === requestNodeIndex))) {
            return this._TemplateRef_4_5;
        }
        if (((token === import11.NgIf) && (4 === requestNodeIndex))) {
            return this._NgIf_4_6;
        }
        if (((token === import12.TemplateRef) && (6 === requestNodeIndex))) {
            return this._TemplateRef_6_5;
        }
        if (((token === import11.NgIf) && (6 === requestNodeIndex))) {
            return this._NgIf_6_6;
        }
        return notFoundResult;
    };
    _View_ModalFooterComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.context.showDefaultButtons;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._NgIf_4_6.ngIf = currVal_0;
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.context.showDefaultButtons;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this._NgIf_6_6.ngIf = currVal_1;
            this._expr_1 = currVal_1;
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_ModalFooterComponent0;
}(import1.AppView));
function viewFactory_ModalFooterComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalFooterComponent === null)) {
        (renderType_ModalFooterComponent = viewUtils.createRenderComponentType('/Users/doug/Source/dougludlow/ng2-bs3-modal/src/ng2-bs3-modal/components/modal-footer.ts class ModalFooterComponent - inline template', 1, import9.ViewEncapsulation.None, styles_ModalFooterComponent, {}));
    }
    return new _View_ModalFooterComponent0(viewUtils, parentInjector, declarationEl);
}
exports.viewFactory_ModalFooterComponent0 = viewFactory_ModalFooterComponent0;
var _View_ModalFooterComponent1 = (function (_super) {
    __extends(_View_ModalFooterComponent1, _super);
    function _View_ModalFooterComponent1(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalFooterComponent1, renderType_ModalFooterComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalFooterComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'button', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'btn btn-default');
        this.renderer.setElementAttribute(this._el_0, 'data-dismiss', 'modal');
        this.renderer.setElementAttribute(this._el_0, 'type', 'button');
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this._expr_1 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1
        ], [disposable_0], []);
        return null;
    };
    _View_ModalFooterComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = import4.interpolate(1, '', this.parent.context.dismissButtonLabel, '');
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setText(this._text_1, currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_ModalFooterComponent1.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.context.modal.dismiss() !== false);
        return (true && pd_0);
    };
    return _View_ModalFooterComponent1;
}(import1.AppView));
function viewFactory_ModalFooterComponent1(viewUtils, parentInjector, declarationEl) {
    return new _View_ModalFooterComponent1(viewUtils, parentInjector, declarationEl);
}
var _View_ModalFooterComponent2 = (function (_super) {
    __extends(_View_ModalFooterComponent2, _super);
    function _View_ModalFooterComponent2(viewUtils, parentInjector, declarationEl) {
        return _super.call(this, _View_ModalFooterComponent2, renderType_ModalFooterComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways) || this;
    }
    _View_ModalFooterComponent2.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'button', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'btn btn-primary');
        this.renderer.setElementAttribute(this._el_0, 'type', 'button');
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this._expr_1 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1
        ], [disposable_0], []);
        return null;
    };
    _View_ModalFooterComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = import4.interpolate(1, '', this.parent.context.closeButtonLabel, '');
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setText(this._text_1, currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_ModalFooterComponent2.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.context.modal.close() !== false);
        return (true && pd_0);
    };
    return _View_ModalFooterComponent2;
}(import1.AppView));
function viewFactory_ModalFooterComponent2(viewUtils, parentInjector, declarationEl) {
    return new _View_ModalFooterComponent2(viewUtils, parentInjector, declarationEl);
}
