// ng2-date-picker/calendar-nav/calendar-nav.component.js
"use strict";
var core_1 = require("@angular/core");
var CalendarNavComponent = (function () {
    function CalendarNavComponent() {
        this.onLeftNav = new core_1.EventEmitter();
        this.onRightNav = new core_1.EventEmitter();
        this.onLabelClick = new core_1.EventEmitter();
        this.isLabelClickable = false;
        this.showLeftNav = true;
        this.showRightNav = true;
        this.leftNavDisabled = false;
        this.rightNavDisabled = false;
    }
    CalendarNavComponent.prototype.leftNavClicked = function () {
        this.onLeftNav.emit();
    };
    CalendarNavComponent.prototype.rightNavClicked = function () {
        this.onRightNav.emit();
    };
    CalendarNavComponent.prototype.labelClicked = function () {
        this.onLabelClick.emit();
    };
    return CalendarNavComponent;
}());
CalendarNavComponent.decorators = [
    { type: core_1.Component, args: [{
                selector: 'dp-calendar-nav',
                template: '<div class="dp-calendar-nav-container">   <button type="button"           class="dp-calendar-nav-left"           [hidden]="!showLeftNav"           [disabled]="leftNavDisabled"           (click)="leftNavClicked()">  &lt;   </button>   <span class="dp-nav-header" [hidden]="isLabelClickable">{{label}}</span>   <button type="button"           class="dp-nav-header dp-nav-header-btn"           [hidden]="!isLabelClickable"           (click)="labelClicked()">     {{label}}   </button>   <button type="button"           class="dp-calendar-nav-right"           [hidden]="!showRightNav"           [disabled]="rightNavDisabled"           (click)="rightNavClicked()">  &gt;   </button> </div> ',
                styles: ['.dp-calendar-nav-container {  position: relative;  box-sizing: border-box;  height: 25px;  border: 1px solid #000000;  border-bottom: none;}.dp-nav-date-btn {  box-sizing: border-box;  height: 25px;  border: 1px solid #000000;  border-bottom: none;}.dp-calendar-nav-left,.dp-calendar-nav-right {  position: absolute;  top: 50%;  transform: translateY(-50%);  cursor: pointer;}.dp-calendar-nav-left {  left: 0;}.dp-calendar-nav-right {  right: 0;}.dp-nav-header {  position: absolute;  top: 50%;  left: 50%;  transform: translate(-50%, -50%);  font-size: 13px;}.dp-nav-header-btn {  cursor: pointer;}:host.dp-material .dp-calendar-nav-container {  height: 30px;  border: 1px solid #E0E0E0;}:host.dp-material .dp-calendar-nav-left,:host.dp-material .dp-calendar-nav-right {  border: none;  background: #FFFFFF;  outline: none;  font-size: 16px;}:host.dp-material .dp-nav-header-btn {  height: 20px;  width: 80px;  border: none;  background: #FFFFFF;  outline: none;}:host.dp-material .dp-nav-header-btn:hover {  background: rgba(0, 0, 0, 0.05);}:host.dp-material .dp-nav-header-btn:active {  background: rgba(0, 0, 0, 0.1);}']
            },] },
];
/** @nocollapse */
CalendarNavComponent.ctorParameters = function () { return []; };
CalendarNavComponent.propDecorators = {
    'onLeftNav': [{ type: core_1.Output },],
    'onRightNav': [{ type: core_1.Output },],
    'onLabelClick': [{ type: core_1.Output },],
    'label': [{ type: core_1.Input },],
    'isLabelClickable': [{ type: core_1.Input },],
    'showLeftNav': [{ type: core_1.Input },],
    'showRightNav': [{ type: core_1.Input },],
    'leftNavDisabled': [{ type: core_1.Input },],
    'rightNavDisabled': [{ type: core_1.Input },],
    'theme': [{ type: core_1.HostBinding, args: ['class',] }, { type: core_1.Input },],
};
exports.CalendarNavComponent = CalendarNavComponent;
